main.go

```diff
@@ -287,10 +292,15 @@ func main() {
 				base = t
 			}
 
+			base.(*http.Transport).DialContext = (&net.Dialer{
```

Comment:
This will have a funny side-effect of modifying the global `http.DefaultTransport`, e.g. even the transport used for OIDC. By convention we tend to clone the transport so that we don't modify the global, e.g. on line 290